File: in_Java/0884 Uncommon Words from Two Sentences.java
class Solution {
    public String[] uncommonFromSentences(String A, String B) {
        List<String> L = new ArrayList<>();
        HashMap<String, Integer> hma = new HashMap<>();
        HashMap<String, Integer> hmb = new HashMap<>();
        for(String s : A.split(" ")){
            if(!hma.containsKey(s)) hma.put(s,1);
            else hma.put(s, hma.get(s)+1);
        }
        for(String s : B.split(" ")){
            if(!hmb.containsKey(s)) hmb.put(s,1);
            else hmb.put(s, hmb.get(s)+1);
        }
        for(String x : hma.keySet()){
            if(hma.get(x) == 1 && !hmb.containsKey(x)) L.add(x);
        }
        for(String x : hmb.keySet()){
            if(hmb.get(x) == 1 && !hma.containsKey(x)) L.add(x);
        }
        String[] res = new String[L.size()];
        int id = 0;
        for(String x : L) res[id++] = x;
        return res;
    }
}

File: in_Python/0884 Uncommon Words from Two Sentences.py
class Solution(object):
    def uncommonFromSentences(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: List[str]
        """
        res = []
        d1 = {}
        d2 = {}
        a = A.split(' ')
        b = B.split(' ')
        for sa in a:
            if sa not in d1:
                d1[sa] = 1
            else:
                d1[sa] += 1
        for sb in b:
            if sb not in d2:
                d2[sb] = 1
            else:
                d2[sb] += 1
        for k in d1:
            if k not in d2 and d1[k] == 1:
                res.append(k)
        for k in d2:
            if k not in d1 and d2[k] == 1:
                res.append(k)
        return res